src/editor.rs

```diff
@@ -476,6 +479,13 @@ impl<'a, W: TermWrite + Write> Editor<'a, W> {
         self.print_current_buffer(false)
     }
 
+    /// Deletes every character preceding the cursor until the beggining of the line
+    pub fn delete_all_before_cursor(&mut self) -> io::Result<()> {
+        cur_buf_mut!(self).remove(0, self.cursor);
+        self.cursor = 0;
+	    self.print_current_buffer(false)
+    }
+
```

Comment:
Is it ok to call `self.cursor = 0` directly or should I use the return value of `cur_buf_mut!(self).remove(0, self.cursor)`?